### PR TITLE
feat(minimax): add MiniMax provider with tier-aware rate limiting

### PR DESCRIPTION
**Pull request by Societus**
## Summary

Add MiniMax as a built-in LLM provider using the generic tier framework from #82.

This PR is a straightforward application of the same pattern as #83. Both MiniMax and Z.AI are OpenAI-compatible APIs with subscription tiers and built-in reasoning models. The generic tier framework made this provider almost mechanical to implement -- the only provider-specific code is the model names, the `reasoning_split` parameter vs Z.AI's `thinking` toggle, and the tier definitions.

**Depends on: #82** (generic tier framework -- merge that first)

## Why This Was Inconsequential

MiniMax shares the same architectural profile as Z.AI:
- OpenAI-compatible API at `https://api.minimax.io/v1`
- Bearer token auth using the same `openai` SDK
- Reasoning models with a thinking separator parameter
- Published subscription tiers with conservative rate limits

The generic framework from #82 eliminated all boilerplate for tier resolution. Adding MiniMax was just: define `RATE_LIMIT_TIERS`, set the base URL, and pick the reasoning parameter name. Everything else is inherited.

### New: MiniMax Provider (`minimax.py`)
- `RATE_LIMIT_TIERS` with Starter/Plus/Max/Ultra configs from published limits
- `resolve_rate_limiter()` from BaseProvider (zero custom tier code)
- `reasoning_split=True` by default (separates thinking from content)
- Retry budget: 5 retries / 30s max wait
- Models: MiniMax-M2.7 (default), M2.7-highspeed, M2.5, M2.5-highspeed, M2.1, M2.1-highspeed, M2

### Registry (`registry.py`)
- Register `minimax` -> `MiniMaxProvider` with `openai` package hint

### Rate Limiter (`rate_limiter.py`)
- `PROVIDER_DEFAULTS["minimax"]` = Starter-tier conservative (5 RPM / 25K TPM)

### CLI Helpers (`helpers.py`)
- `MINIMAX_API_KEY`, `MINIMAX_BASE_URL`, `MINIMAX_REASONING_SPLIT`, `MINIMAX_TIER` env vars
- Auto-detect from `MINIMAX_API_KEY`
- Added to provider validation list

### Tests (`test_minimax_provider.py`)
- 30 tests: constructor, tier resolution (4 tiers + edge cases), generate with mock, stream_chat, reasoning_split, registry integration

## Rate Limit Tiers

From published MiniMax docs (5-hour rolling window):

| Tier | Requests/5hrs | RPM | TPM |
|------|--------------|-----|-----|
| Starter | 1,500 | 5 | 25,000 |
| Plus | 4,500 | 15 | 75,000 |
| Max | 15,000 | 50 | 250,000 |
| Ultra | 30,000 | 100 | 500,000 |

Highspeed variants (e.g., MiniMax-M2.7-highspeed) share the same rate limits as their base plan. The difference is model selection (faster inference), not quota.

Ref: https://platform.minimax.io/docs/token-plan/intro

## Configuration

```bash
export MINIMAX_API_KEY="***"
export MINIMAX_TIER="plus"              # starter | plus | max | ultra
export MINIMAX_BASE_URL="..."           # override default
export MINIMAX_REASONING_SPLIT="true"   # default true
```

## Test Plan

```bash
uv run pytest tests/unit/test_providers/test_minimax_provider.py -v
# 30 passed
```

All 121 provider tests pass with zero regressions.

## PR Stack

| # | PR | Status |
|---|-----|--------|
| 1 | **#82 -- Generic tier framework** | Ready for review |
| 2 | **#83 -- Z.AI adopts the framework** | Depends on #82 |
| 3 | **This PR -- MiniMax provider** | Depends on #82 |

## Related

- Ref: #68

### CODE REVIEW

File: packages/cli/src/repowise/cli/ui.py [hunk 1]
```diff
@@ -476,14 +484,22 @@ def interactive_provider_select(
...
-        console.print(f"  [bold]{chosen}[/bold] requires [cyan]{env_var}[/cyan].")
-        if signup_url:
-            console.print(f"  Get your API key here: [{BRAND}]{signup_url}[/]")
-        console.print()
-        key = _prompt_api_key(console, chosen, env_var, repo_path=repo_path)
-        if not key:
-            console.print(f"  [{WARN}]Skipped. Please select another provider.[/]")
-            return interactive_provider_select(console, model_flag, repo_path=repo_path)
+        # Special case: litellm local proxy doesn't need an API key
+        if chosen == "litellm" and os.environ.get("LITELLM_BASE_URL"):
```
Comment (swati510) [inline packages/cli/src/repowise/cli/ui.py]: this branch is unreachable — _detect_provider_status (L417-420) already marks   litellm as detected when LITELLM_BASE_URL is set, so we never enter the outer if chosen not in detected with this combo.

File: packages/cli/src/repowise/cli/ui.py [hunk 2]
```diff
@@ -268,18 +268,22 @@ def print_phase_header(
     "litellm": "groq/llama-3.1-70b-versatile",
 }
```
Comment (swati510) [inline packages/cli/src/repowise/cli/ui.py:271]: zai and minimax are wired in helpers.py, validate_provider_config, and the       registry but not here ,  they won't show up in the interactive init menu. please add them to _PROVIDER_DEFAULTS, _PROVIDER_ENV, and          _PROVIDER_SIGNUP

File: packages/core/src/repowise/core/providers/llm/zai.py [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+                  over plan selection.
+        thinking: Thinking mode for GLM-5 family. 'disabled' by default
+                  to avoid reasoning token overhead. Set to 'enabled' for
+                  complex reasoning tasks.
+        rate_limiter: Optional RateLimiter instance.
+        cost_tracker: Optional CostTracker for usage tracking.
+    """
+
+    def __init__(
+        self,
```
Comment (swati510) [inline packages/core/src/repowise/core/providers/llm/zai.py:99]: since this PR introduces the tier framework on BaseProvider, should zai adopt it  too? lite/pro/max have published limits. ok to defer but feels odd to land the framework and only wire minimax

**Review by swati510 — COMMENTED** [review]
missing zai and minimax in providers/llm/__init__.py, registry.py docstring got updated, it didnt

**Review by swati510 — COMMENTED** [review]
Looks like this is stacked on #83, so the base.py/registry/zai changes are shared. Assuming #83 lands first this is fine, just calling it out.

Three things:

1. My earlier note about _PROVIDER_DEFAULTS / _PROVIDER_ENV / _PROVIDER_SIGNUP in cli/ui.py still stands, zai and minimax are invisible in the interactive init menu. Worth fixing here since this PR ships both.

2. MiniMax rate limits are published as 1500 requests / 5 hours. Our RateLimiter is a 60-second sliding window. Converting to ~5 RPM is a reasonable steady-state approximation but a user who bursts will see spurious 429s locally, and one who paces slowly can technically exceed quota without tripping our limiter. Fine to ship as-is, but leave a comment acknowledging the window mismatch so nobody chases a ghost bug later.

3. MINIMAX_REASONING_SPLIT is parsed as .lower() == "true" in two different branches of helpers.py. Extract a tiny _env_bool helper and accept the usual truthy values (1/yes/on) since that's what users reach for.

File: packages/cli/src/repowise/cli/helpers.py [hunk 4]
```diff
@@ -262,6 +262,32 @@ def resolve_provider(
...
+            if os.environ.get("ZAI_THINKING"):
+                kwargs["thinking"] = os.environ["ZAI_THINKING"]
+        elif provider_name == "minimax":
+            # MiniMax: API key, base URL, reasoning_split, and tier
+            if os.environ.get("MINIMAX_API_KEY"):
+                kwargs["api_key"] = os.environ["MINIMAX_API_KEY"]
+            if os.environ.get("MINIMAX_BASE_URL"):
+                kwargs["base_url"] = os.environ["MINIMAX_BASE_URL"]
+            if os.environ.get("MINIMAX_REASONING_SPLIT"):
+                kwargs["reasoning_split"] = os.environ["MINIMAX_REASONING_SPLIT"].lower() == "true"
```
Comment (swati510) [inline packages/cli/src/repowise/cli/helpers.py]: Same .lower() == "true" parsing is copy-pasted at line 357 in the auto-detect path. Extract into _env_bool(name, default=False) and reuse. Also accept 1/yes/on, that's what users will type.

**Comment by RaghavChamadiya** [comment]
This is currently stacked on #83 and re-includes all of #83's framework code verbatim. Once #83 lands, please rebase so this PR shrinks to just the MiniMax-specific delta (provider, env vars, registry entry, tests). It will be much easier to review at that point.

While you're rebasing, please also pick up the four items below so we can close everything out in one shot:

1. Missing exports. `zai` and `minimax` aren't added to `packages/core/src/repowise/core/providers/llm/__init__.py`, so they're invisible to consumers importing from the package root.

2. Interactive init menu. `_PROVIDER_DEFAULTS`, `_PROVIDER_ENV`, and `_PROVIDER_SIGNUP` in `packages/cli/src/repowise/cli/ui.py` don't list either provider, so neither shows up in the interactive provider picker. Worth adding both here since this PR ships them.

3. Rate-limit window mismatch. MiniMax publishes 1500 req / 5 hr, but our `RateLimiter` is a 60-second sliding window. Converting to ~5 RPM is a reasonable steady-state approximation but a user who bursts will see spurious 429s locally and one who paces slowly can technically exceed quota without tripping the limiter. Fine to ship as-is, just leave a code comment in `minimax.py` acknowledging the window mismatch so nobody chases a ghost bug later.

4. `MINIMAX_REASONING_SPLIT` is parsed as `.lower() == "true"` in two different branches of `helpers.py`. Extract a tiny `_env_bool` helper and accept the usual truthy values (`1`, `yes`, `on`, `true`) since that's what users reach for.

Once #83 is merged and this is rebased + the four items above are addressed, this is ready to land.

File: packages/cli/src/repowise/cli/ui.py (hunk 2 above)
Comment (Societus) [reply packages/cli/src/repowise/cli/ui.py:271]: > zai and minimax are wired in helpers.py, validate_provider_config, and the registry but not here , they won't show up in the interactive init menu. please add them to _PROVIDER_DEFAULTS, _PROVIDER_ENV, and _PROVIDER_SIGNUP

Plan so far is to make changes after 83 finalizes, will address everything in one go, time has been short the last week.
